src/fides/api/ops/api/v1/endpoints/user_endpoints.py

```diff
@@ -191,3 +211,120 @@ def user_logout(
     logger.info("Logging out user.")
     if client:
         client.delete(db)
+
+
+@router.put(
+    urls.SYSTEM_MANAGER,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_UPDATE])],
+    response_model=List[SystemSchema],
+)
+def update_system_manager(
+    *,
+    db: Session = Depends(deps.get_db),
+    user_id: str,
+    systems: List[FidesKey],
+) -> List[SystemSchema]:
+    """
+    Endpoint to override the systems for which a user is "system manager".
+    All systems the user manages are replaced with those in the request body.
+    """
+    user = validate_user_id(db, user_id)
+
+    if len(set(systems)) != len(systems):
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Cannot add user {user_id} as system manager. Duplicate systems in request body.",
+        )
+
+    retrieved_systems: Query = db.query(System).filter(System.fides_key.in_(systems))
+    if retrieved_systems.count() != len(systems):
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"Cannot add user {user_id} as system manager. System(s) not found.",
+        )
+
+    logger.info("Updating systems for which user {} is system manager", user_id)
+
+    # Adding new systems for which the user is not already a manager
+    for system in retrieved_systems:
+        if user not in system.users:
+            user.set_as_system_manager(db, system)
+
+    # Removing systems for which the user in no longer a manager
+    for system in user.systems:
+        if system not in retrieved_systems:
+            user.remove_as_system_manager(db, system)
+
+    return user.systems
+
+
+@router.get(
+    urls.SYSTEM_MANAGER,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_READ])],
+    response_model=List[SystemSchema],
+)
+def get_user_systems(
+    *,
+    db: Session = Depends(deps.get_db),
+    user_id: str,
+) -> List[SystemSchema]:
+    """
+    Endpoint to retrieve all the systems for which a user is "system manager".
+    """
+    user = validate_user_id(db, user_id)
+    logger.info("Getting systems for which user {} is system manager", user_id)
+
+    return user.systems
+
+
+@router.get(
+    urls.SYSTEM_MANAGER_DETAIL,
+    dependencies=[Security(verify_oauth_client, scopes=[SYSTEM_MANAGER_READ])],
+    response_model=SystemSchema,
+)
+def get_user_system(
+    *, db: Session = Depends(deps.get_db), user_id: str, system_key: FidesKey
+) -> SystemSchema:
+    """
+    Endpoint to retrieve a single system managed by the given user.
+    """
```

Comment:
how would this endpoint be used differently from `/api/v1/system/{fides_key}`? I'd expect `/api/v1/system/{fides_key}` to also either 404 or return the system if the user manages it?

Reply:
If you had permission to view the system in any capacity, I'd think `/api/v1/system/{fides_key}` should return the system, unrelated to whether you are a manager.

This extra endpoint makes the distinction of 404'ing if the user is not a manager.  I added in case it was useful, but that may not be helpful here.  We could just use `/api/v1/system/{fides_key}` to retrieve an individual system.  Should I remove it @allisonking?

Reply:
ah got it! let's keep it for now 👍